floodviz/static/js/hydrograph.js

```diff
@@ -1,149 +1,183 @@
 				svg.append("g")
 					.attr('class', 'gages')
 					.append("path")
 					.attr("id", "hydro" + d.key)
 					.attr("d", line(d.values));
 			});
-
 			// Add the X Axis
 			svg.append("g")
 				.attr("class", "axis")
 				.attr("transform", "translate(0," + height + ")")
 				.call(d3.axisBottom(x).tickFormat(d3.timeFormat("%B %e")));
-
 			// Add the Y Axis
 			svg.append("g")
 				.attr("class", "axis")
 				.call(d3.axisLeft(y).ticks(10, ".0f"));
-
-			var focus = svg.append("g")
-				.attr("transform", "translate(-100,-100)")
-				.attr("class", "focus");
+			// Tooltip
+			focus = svg.append("g")
+					.attr("transform", "translate(-100,-100)")
+					.attr("class", "focus");
 			focus.append("circle")
-				.attr("r", 3.5);
-
+					.attr("r", 3.5);
 			focus.append("text")
-				.attr("y", -10);
-
-			var voronoiGroup = svg.append("g")
-				.attr("class", "voronoi");
-
-			voronoiGroup.selectAll("path")
-				.data(voronoi.polygons(d3.merge(dataNest.map(function (d) {
-					return d.values
-				}))))
-				.enter().append("path")
-				.attr("d", function (d) {
-					return d ? "M" + d.join("L") + "Z" : null;
-				})
-				.on("mouseover", mouseover)
-				.on("mouseout", mouseout)
-				.on("click", function(d) { click(d.data.key, graph_data); });
-
-			function mouseover(d) {
-				d3.select(d.data.name).classed("gage--hover", true);
-				focus.attr("transform", "translate(" + x(d.data.time_mili) + "," + y(d.data.value) + ")");
-				focus.select("text").html(d.data.key + ": " + d.data.value + " cfs " + " " + d.data.time + " " + d.data.timezone);
-			}
-
-			function mouseout(d) {
-				d3.select(d.data.name).classed("gage--hover", false);
-				focus.attr("transform", "translate(-100,-100)");
-			}
-		}
-
-		function click(key, data) {
-
+					.attr("y", -10);
+			// Voronoi Layer
+			voronoi_group = svg.append("g")
+					.attr("class", "voronoi");
+			voronoi_group.selectAll("path")
+					.data(voronoi.polygons(d3.merge(dataNest.map(function (d) {
+						return d.values
+					}))))
+					.enter().append("path")
+					.attr("d", function (d) {
+						return d ? "M" + d.join("L") + "Z" : null;
+					})
+					.on("mouseover", self.mouseover)
+					.on("mouseout", self.mouseout)
+					.on("click", function(d) {return self.click(d.data.key, graph_data)})
+		};
+    
+		/**
+		 * Initalize the Hydrograph
+		 */
+		self.init = function() {
+			var data_path = options.data_path;
+			// Get the data
+			d3.json(data_path, function (error, data) {
+				update(data);
+			});
+		};
+    
+		/**
+		 * Handle all mouse over movements for calling element. 
+		 */
+		self.mouseover = function(d) {
+			d3.select(d.data.name).classed("gage--hover", true);
+			focus.attr("transform", "translate(" + x(d.data.time_mili) + "," + y(d.data.value) + ")");
+			focus.select("text").html(d.data.key + ": " + d.data.value + " cfs " + " " + d.data.time + " " + d.data.timezone);
+			// Interative linking with map
+			FV.map_figure.mousemove(d.data.name, d.data.key);
```

Comment:
This should not be here. This is going to cause problems when the map_figure call hydrgraph to do the highlighting. This line should move to inside the mouseover event handler.